### PR TITLE
Updates to MT broker docs

### LINKED ISSUE #2345: Update docs to include changes related to Multi Tenant channel based broker
## What is changing? (Please include as many details as possible.)

As of 0.14, MT channel based broker will be introduced
https://knative.dev/development/eventing/mt-channel-based-broker/

## How will this impact our users?
Knative will now provide a Multi Tenant Broker implementation that uses Channels for event routing. 

## In what release will this take happen (to the best of your knowledge)?
0.14

## Context
Link to associated PRs or issues from other repos here.

1. https://github.com/knative/docs/pull/2321 previous work done by @vaikas 
2. Doc with context: https://docs.google.com/document/d/1qYnmkRduWLUFQ3vEsaw7jU_mxS_nDHHkDkcGRf1_Fy4/edit#heading=h.ra3is5gwyzel

## Additional info
Requirements for docs are as follows:

1. Going through the other docs to see what needs to be revised in light of the MT broker.
2. Going over the MT broker docs with a tech writer eye.

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes #2345 
Tech writer review of docs created by @vaikas [WIP]

### CODE REVIEW

File: docs/eventing/channel-based-broker.md [hunk 1]
```diff
@@ -53,6 +58,18 @@ data:
...
 
+**NOTE:** This example specifies two extra parameters that are specific to Kakfa Channels; `numPartitions` and `replicationFactor`.
+
+## Configuring the MT Broker
+
+After you have configured the ConfigMap for each Channel provider type, you can configure the MT Broker.
+You can configure which Channels are used as a cluster level default, by namespace or for a specific Broker.
+The Channels used are configured by a `config-br-defaults` ConfigMap in the `knative-eventing` namespace.
+
+The following example ConfigMap uses a Kafka channel for all Brokers, except for the `test-broker-6` namespace,  which uses InMemoryChannel.
```
Comment (vaikas) [inline docs/eventing/channel-based-broker.md]: maybe change test-broker-6 to something else, like imc-namespace or something like that. this was just one of the test namespaces I was using while writing the docs :) 
And if you do that, it has ripple effects after this point.

**Comment by mattmoor** [comment]
/approve
/hold

Approving to trigger stage-docs

File: docs/eventing/broker/README.md [hunk 2]
```diff
@@ -57,6 +57,56 @@ spec:
...
+### Creating a broker by annotation
+
+You can create a broker annotate your namespace:
+
+```shell
+kubectl label namespace <namespace> knative-eventing-injection=enabled
+```
+
+This will automatically create a broker named `default`, which is configured to use
+Kafka channels, in the specified namespace.
```
Comment (vaikas) [inline docs/eventing/broker/README.md:69]: unless the default-br-channel has been somewhere else to be configured to use Kafka as default, then this is not exactly right?
So, here we show how to configure to use Kafka channels for a certain namespace (test-broker-6, cause that's what I was using for my testing :) )
https://github.com/knative/docs/blob/master/docs/eventing/broker/mt-channel-based-broker.md

So, maybe just drop the Kafka channel here, and add another section that specifically talks about how to configure default channels?

So, out of the box, simple example, uses, say IMC as the default channel. Then having a different section for configuring the defaulting behaviour?

Also, given how easy it is now to create brokers, I'm wondering if we should first just have a user create a Broker using a kubectl?

File: docs/eventing/broker/mt-channel-based-broker.md [hunk 3]
```diff
@@ -4,27 +4,32 @@ weight: 30
...
+
+**NOTE:** This guide assumes Knative Eventing is installed in the `knative-eventing`
+namespace. If you have installed Knative Eventing in a different namespace, replace
+`knative-eventing` with the name of that namespace.
+
+## Configure Channel ConfigMaps
+
+You can define specifications for how each type of channel will be created, by modifying the ConfigMap for each channel type.
+
+<!-- TODO: Split these configmaps out and document them properly in a section for each channel, then just link from here-->
```
Comment (vaikas) [inline docs/eventing/broker/mt-channel-based-broker.md:26]: Ah, I see you do it here :) 👍

File: docs/eventing/broker/mt-channel-based-broker.md [hunk 4]
```diff
@@ -54,6 +62,20 @@ data:
       replicationFactor: 1
 ```
 
+**NOTE:** This example specifies two extra parameters that are specific to Kakfa Channels; `numPartitions` and `replicationFactor`.
```
Comment (vaikas) [inline docs/eventing/broker/mt-channel-based-broker.md:65]: nit: Kakfa => Kafka

File: docs/eventing/broker/mt-channel-based-broker.md [hunk 5]
```diff
@@ -70,17 +92,16 @@ data:
       namespace: knative-eventing
     namespaceDefaults:
       brokerClass: MTChannelBasedBroker
-      test-broker-6:
+      example-ns:
```
Comment (vaikas) [inline docs/eventing/broker/mt-channel-based-broker.md:95]: Thanks :)

File: docs/eventing/broker/README.md [hunk 6]
```diff
@@ -57,6 +57,56 @@ spec:
         name: dlq-service
 ```
 
+### Creating a broker by annotation
+
+You can create a broker annotate your namespace:
```
[the remark this answers is not in the document]
Comment (abrennan89) [reply docs/eventing/broker/README.md:62]: "using annotation" I guess :)

File: docs/eventing/broker/mt-channel-based-broker.md [hunk 7]
```diff
@@ -54,6 +62,20 @@ data:
...
+
+## Configuring the MT broker
+
+After you have configured the ConfigMap for each Channel provider type, you can configure the MT Broker.
+
+Channels can be configured as a cluster level default, by namespace, or for a specific broker.
+
+The channels used by the MT broker can be configured in the `config-br-defaults` ConfigMap in the `knative-eventing` namespace.
+
+The following example ConfigMap uses a Kafka channel for all brokers, except for the `example-ns` namespace,  which uses InMemoryChannel.
```
Comment (vaikas) [inline docs/eventing/broker/mt-channel-based-broker.md:75]: super nit: extra space after comma after namespace.

File: docs/eventing/broker/README.md (hunk 2 above)
Comment (abrennan89) [reply docs/eventing/broker/README.md:69]: I would say this is all stuff we should follow up in our subsequent PRs to fix up this section - there's a lot of work to do to fix channels too, period.

**Pull request merged**